tangelo/tangelo/__init__.py

```diff
@@ -126,9 +130,8 @@ def good(path):
             log("Illegal path (absolute): %s" % (orig), "SERVICE")
             return None
 
-        path = os.path.abspath(cherrypy.thread_data.modulepath + os.path.sep +
-                               path)
-        if len(path) >= len(root) and path[:len(root)] == root:
+        path = os.path.abspath(os.path.join(cherrypy.thread_data.modulepath, path))
+        if path == root or path.startswith(root + os.path.sep):
```

Comment:
Great fix!  Thanks for simplifying the ``os.path`` code - I was in the midst of learning that module when I originally wrote this :-D.